persimmon/website/views.py

```diff
@@ -133,3 +144,49 @@ def get_my_accounts(request):
         'balance': account.balance,
         'approval_status': account.approval_status,
     } for account in accounts]
+
+
+@api_function
+def approve_credit_debit_funds(request, account_number: int, approved: bool, credittype0debittype1: int):
+    current_user(request, required_auth=EmployeeLevel.TELLER)
+    try:
+        pendingtransactions = BankStatements \
+            .objects.filter(bankAccountId=account_number, approval_status=ApprovalStatus.PENDING)
+    except BankStatements.DoesNotExist as exc:
+        raise Http404("No such transaction pending approval") from exc
+    for creditdebit in pendingtransactions:
+        if approved:
+            creditdebit.approve_status = ApprovalStatus.APPROVED
+            account = BankAccount.objects.filter(id=account_number)
+            if credittype0debittype1 == 0:
+                account.balance -= int(creditdebit.transaction[1])
+            else:
+                account.balance += int(creditdebit.transaction[1])
```

Comment:
You need to `save()` the account after modifying it.

Reply:
This is a situation where you're performing multiple actions that need to be done atomically on the database to prevent race conditions. You can do this with what's called a db transaction. See my comment on Brennan's PR for more details.

Reply:
ok

Reply:
should I just add an annotation to the whole function like this:
from django.db import transaction
@transaction.atomic
def viewfunc(request):
    # This code executes inside a transaction.
    do_stuff()

cause if using the below may need to happen multiple places, and I do not know where to insert it.
from django.db import transaction

def viewfunc(request):
    # This code executes in autocommit mode (Django's default).
    do_stuff()

    with transaction.atomic():
        # This code executes inside a transaction.
        do_more_stuff()

Reply:
Making the entire function atomic may incur more of a performance overhead than strictly necessary, but it will certainly get the job done. If it causes performance issues we can deal with them down the line. (I suspect we will not need to).